ocs_ci/ocs/registry.py

```diff
@@ -0,0 +1,209 @@
+import logging
+import time
+import base64
+
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.resources import pod
+from ocs_ci.utility.utils import run_cmd
+from tests import helpers
+
+
+logger = logging.getLogger(__name__)
+
+
+def change_registry_backend_to_ocs():
+    """
+    Function to deploy registry with OCS backend.
+    """
+    pv_obj = helpers.create_pvc(
+        sc_name=constants.DEFAULT_SC_CEPHFS, pvc_name='registry-cephfs-rwx-pvc',
+        namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE, size='100Gi',
+        access_mode=constants.ACCESS_MODE_RWX
+    )
+    helpers.wait_for_resource_state(pv_obj, 'Bound')
+    ocp_obj = ocp.OCP(
+        kind=constants.CONFIG, namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+    )
+    param_cmd = f'[{{"op": "add", "path": "/spec/storage", "value": {{"pvc": {{"claim": "{pv_obj.name}"}}}}}}]'
+    assert ocp_obj.patch(
+        resource_name=constants.IMAGE_REGISTRY_RESOURCE_NAME, params=param_cmd
+    ), f"Registry pod storage backend to OCS is not success"
+
+    # Validate registry pod status
+    validate_registry_pod_status()
+
+    # Validate pvc mount in the registry pod
+    validate_pvc_mount_on_registry_pod()
+
+
+def get_registry_pod_obj():
+    """
+    Function to get registry pod obj
+    Return:
+        pod_obj (obj): Registry pod obj
+    """
+    # Sometimes when there is a update in config crd, there will be 2 registry pods
+    # i.e. old pod will be terminated and new pod will be up based on new crd
+    # so below loop waits till old pod terminates
+    wait_time = 30
+    while True:
```

Comment:
while True is danger loop if you don't have specified the number of retries or timeout inside! Please when you use while True you should always do it otherwise can be here fore infinite time when some unexpected behavior happens.

Reply:
ack, WIll add more checks to break the loop.

Reply:
Updated loop with more check to exit.

Reply:
Isn't than better option then just use for iteration in range(30):

you can then remove lines:
iteration = 0
iteration += 1

Reply:
ack